crates/sncast/src/lib.rs

```diff
@@ -278,15 +279,27 @@ pub async fn wait_for_tx(
     }
     let retries = timeout / u16::from(retry_interval);
     for i in (1..retries).rev() {
-        match provider.get_transaction_receipt(tx_hash).await {
-            Ok(receipt) => {
-                return match receipt.execution_result() {
-                    ExecutionResult::Succeeded => Ok("Transaction accepted"),
-                    ExecutionResult::Reverted { reason } => {
-                        Err(anyhow!("Transaction has been reverted = {reason}"))
+        match provider.get_transaction_status(tx_hash).await {
+            Ok(status) => match status {
+                starknet::core::types::TransactionStatus::Received => {
+                    return Ok("Transaction has been received");
+                }
+                starknet::core::types::TransactionStatus::Rejected => {
+                    return Err(anyhow!("Transaction has been rejected"));
+                }
+                starknet::core::types::TransactionStatus::AcceptedOnL2(execution_status)
+                | starknet::core::types::TransactionStatus::AcceptedOnL1(execution_status) => {
+                    match execution_status {
+                        starknet::core::types::TransactionExecutionStatus::Succeeded => {
+                            return Ok("Transaction accepted")
+                        }
+
+                        starknet::core::types::TransactionExecutionStatus::Reverted => {
+                            return Err(anyhow!("Transaction has been reverted"))
+                        }
```

Comment:
Are you sure this case is possible? I mean `AcceptedOnL2`/`AcceptedOnL1` + `Reverted`

Reply:
```
pub enum TransactionStatus {
    Received,
    Rejected,
    AcceptedOnL2(TransactionExecutionStatus),
    AcceptedOnL1(TransactionExecutionStatus),
}
```
I based on this starknet-rs enum

Reply:
![Zrzut ekranu 2023-12-22 o 14 39 47](https://github.com/foundry-rs/starknet-foundry/assets/15279784/27bdb3ed-c140-489b-b347-da93087eaf3c)